### PR TITLE
simplify get_url_base abstract Sites.objects.get

### PR DESCRIPTION
**Pull request by dannyroberts**

### CODE REVIEW

File: dimagi/utils/web.py [hunk 1]
```diff
@@ -16,18 +20,19 @@
...
-    address = getattr(settings, "BASE_ADDRESS", None)
-    if address is None:
-        address = Site.objects.get(id = settings.SITE_ID).domain
-    return '%s://%s' % (protocol, address)
-
-def get_secure_url_base():
-    return 'https://%s' % Site.objects.get(id = settings.SITE_ID).domain
+        warnings.warn("Please don't use Site to set domain anymore; "
+                      "use BASE_ADDRESS instead.", DeprecationWarning)
+        return Site.objects.get(id=settings.SITE_ID).domain
```
Comment (czue) [inline dimagi/utils/web.py]: is this documented anywhere?

Comment (dannyroberts) [inline dimagi/utils/web.py]: I'm not sure what you mean. I was thinking that if I get rid of all references to `Site` in this fashion throughout the codebase, and keep this here as a failover that displays a deprecation warning, people would be able to fix this for themselves as they notice it. We already basically require `BASE_ADDRESS` for a bunch of other things, and for that to be identical to what was in the `Site` model, so likely most people already have it.

Comment (dannyroberts) [inline dimagi/utils/web.py]: BASE_ADDRESS is also referenced [here](https://github.com/dimagi/commcare-hq/blob/e648657dc663cccea414da8df106a26e851b255b/corehq/apps/builds/README.md#make-commcare-use-this-ip-address) and in a [testrunner error message](https://github.com/dimagi/commcare-hq/blob/e648657dc663cccea414da8df106a26e851b255b/testrunner.py#L125) and in [localsettings.example.py](https://github.com/dimagi/commcare-hq/blob/e648657dc663cccea414da8df106a26e851b255b/localsettings.example.py#L128)

Comment (czue) [inline dimagi/utils/web.py]: sorry - i was asking whether we explicitly tell you to set the site somewhere and thinking we should change that to reference `BASE_ADDRESS`. I thought that existed at some point but possibly we've already fixed it.

Comment (dannyroberts) [inline dimagi/utils/web.py]: Oh, no you're right. I looked to make sure `BASE_ADDRESS` is (somewhat, minimally) documented, but I should also remove instructions about adding `Site`.

Comment (dannyroberts) [inline dimagi/utils/web.py]: Took a pass at it here: https://github.com/dimagi/commcare-hq/pull/5394

Comment (czue) [inline dimagi/utils/web.py]: thanks!

**Pull request merged**